### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by Asgarrrr** [SuspiciousLookingOwl/youtubei#32]
Hi !

**Describe the bug**
Getting information from the mixes does not seem to work 

**To Reproduce**
<img width="720" alt="Capture d’écran 2021-09-26 à 16 01 07" src="https://user-images.githubusercontent.com/36574651/134811085-53302a8b-2031-4800-8896-829e6dc78eb3.png">

**Expected behavior**
Have information about the mix / videos contained in it

**Screenshots**

Full URL of the mix :
<img width="741" alt="Capture d’écran 2021-09-26 à 16 02 25" src="https://user-images.githubusercontent.com/36574651/134811121-0c41cd86-d022-40ba-b16e-613ad88a7c87.png">

The code
<img width="741" alt="Capture d’écran 2021-09-26 à 16 02 52" src="https://user-images.githubusercontent.com/36574651/134811133-31b7db66-a7fc-4379-ab3b-04c82f80a625.png">

I'm now on version `0.0.1-rc.24`

**Comment by SuspiciousLookingOwl** [comment]
Just did some extensive testing, seems like for some reason this particular type of playlist requires one extra data to send to YouTube which is `visitorData` that can only be obtained by sending the request to the YouTube page first.

TL;DR is it's a bit complex and gonna take a while to implement, I probably won't add this anytime soon.

Keep in mind that this type of playlist will contains different videos (or in different order) every time it fetched (you can try this by refreshing the mix playlist page).

I will convert this issue to *Enhancement* instead of *Bug*, thanks for reporting.

**Comment by KagChi** [comment]
> Just did some extensive testing, seems like for some reason this particular type of playlist requires one extra data to send to YouTube which is visitorData that can only be obtained by sending the request to the YouTube page first.

i dont think so that really need, I think some mix playlists require a videoId in the request.

explanation:
- With videoId:
```json
{ "context": {
                "client": {
                    "clientName": "ANDROID",
                    "clientVersion": "16.24"
                }
            },
   "playlistId": "RDTMAK5uy_nilrsVWxrKskY0ZUpVZ3zpB0u4LwWTVJ4",
   "videoId": "40taAdds6AY"
 }
```
post to `https://www.youtube.com/youtubei/v1/next` youtube will give you the data 

- Without videoId:
```json
{ "context": {
                "client": {
                    "clientName": "ANDROID",
                    "clientVersion": "16.24"
                }
            },
   "playlistId": "RDTMAK5uy_nilrsVWxrKskY0ZUpVZ3zpB0u4LwWTVJ4"
   
 }````

post to `https://www.youtube.com/youtubei/v1/next` 

Response: https://bin.zhycorp.net/acofariper.json

**Comment by SuspiciousLookingOwl** [comment]
Ah you might be right, I just tried it again with `videoId` in the body and it worked. Also seems like YouTube doesn't really care on which `videoId` you passed on, but it will affects the videos in the playlist, e.g. passing `dQw4w9WgXcQ` will returns:
```
Rick Astley Never gonna give you up 1 hour seamless loop
a-ha - Take On Me (Official Video) [Remastered in 4K]
George Michael - Careless Whisper (Official Video)
Bee Gees - Stayin' Alive (Official Music Video)
Back to the 80s - Greatest Hits 80s  - Best Oldies Songs Of 1980s - Hits Of The 80s
...
```
It returns very similar result from getting related videos of `dQw4w9WgXcQ`
```js
const v = await youtube.getVideo("dQw4w9WgXcQ");
console.log(v.related.map((v) => v.title));
```

I will probably just separate it to `.getMixPlaylist(playlistId, videoId)` then. However I'm still not really sure why your PR (#29) works for some mix playlist lol. This makes me confused of how Youtube's Mix Playlist actually works 👀

**Comment by SuspiciousLookingOwl** [comment]
Also calling `.next()` in mix playlists throws an error instead of returning an empty array

**Comment by thomasperge** [comment]
Hey, go check out: https://github.com/thomasperge/youtube-scraping-transcription
